### ISSUE TITLE
DCO Expected — Waiting for status to be reported 

### ISSUE THREAD

**Issue by vasukulkarni** [probot/dco#162]
Is there an issue with DCO bot? It seems to be stuck forever in our PR checks?

**Comment by dholbach** [comment]
We were expecting similar issues in the `fluxcd` organisation, e.g. https://github.com/fluxcd/website/pull/580 - we just had to disable the check.

**Comment by brianwarner** [comment]
@gr2m are you able to clarify, are servers being shut off for probot apps in general, or just for apps that don't currently have maintainers?

**Comment by dirien** [comment]
@gr2m, What is the process to take over the ownership? My company can do this, if its okay?

**Comment by gr2m** [comment]
> @gr2m are you able to clarify, are servers being shut off for probot apps in general, or just for apps that don't currently have maintainers?

In general.

The only app in the @probot org that has a maintainer is the settings app. We will move https://github.com/probot/settings/ out of the @probot org and deploy it to servers not owned by GitHub.

> @gr2m, What is the process to take over the ownership? My company, i am working at, can do this, if its okay? We use the DCO too in our OSS projects.

I don't think there is a process, so we can make it up as we go. I'll try to have someone from GitHub reach out, thank you for offering!

**Comment by ryjones** [comment]
Hi, @gr2m , I work for The Linux Foundation and we would be very interested in helping with ownership/maintenance - please send email to community-architects@hyperledger.org

**Comment by martinwoodward** [comment]
@karasowles will get a call set up with @ryjones and @rginn to figure this out and then we can document the plan in this repo.

**Comment by hiimbex** [comment]
I am very disheartened to see GitHub abandon running these apps, especially so suddenly and without any warning. I think that was a horribly unfair move @github. I don't have interest in continuing to maintain this project longterm, but I will step in to help if needed to transfer app/repo ownership.

@gr2m proposal is a great one (transferring the app to a new owner). One alternative I'll throw out there would be to just start fresh with a new app. By that I mean have the Linux Foundation move this repo somewhere they own, and create and host a new app. This would mean the following
1. Someone would need to create a new App (new App ID, private key, etc.) - it would need to be named slightly differently 
1. Host it somewhere
1. Everyone currently using would need to install the app again 
1. Everyone uninstalls the old app 

This way there is no code change/check required to ensure folks are consenting to the new owners, and that way the folks running the app can directly manage it and utilize things like error logs to make improvements. Maybe re-installing the app everywhere is too much work, but considering org level installs, it's possible it might be simpler.

For everyone's reference if creating a new App is something you all might consider, here's the current config for the app:

<img width="388" alt="Screen Shot 2021-10-12 at 5 37 06 PM" src="https://user-images.githubusercontent.com/13410355/137047455-f437ee73-45d0-4fa5-a5ea-9a6c2713eab2.png">
<img width="527" alt="Screen Shot 2021-10-12 at 5 36 53 PM" src="https://user-images.githubusercontent.com/13410355/137047454-ea6fe32d-603e-45b4-816d-fd1e6092ab08.png">
<img width="517" alt="Screen Shot 2021-10-12 at 5 36 46 PM" src="https://user-images.githubusercontent.com/13410355/137047452-89be7f37-b25b-44b1-91bf-3b78b02f912e.png">
<img width="536" alt="Screen Shot 2021-10-12 at 5 36 40 PM" src="https://user-images.githubusercontent.com/13410355/137047448-d6a8f175-c80b-41d1-ad0d-357247a1ecf9.png">

Maybe re-installing the app everywhere is too much work, but considering org level installs, it's possible it might be simpler.

**Comment by shailesh-vaidya** [comment]
We are also facing this issue for our project.

**Comment by mowies** [comment]
Under the `keptn` github organization I also created a DCO check pipeline that can be directly used through [this GH feature](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows)

Link to the pipeline: https://github.com/keptn/gh-automation/blob/main/.github/workflows/dco.yml

An example usage of this pipeline:
```yaml
name: DCO
on:
  pull_request:
  push:
    branches:
      - "master"
jobs:
  check:
    uses: keptn/gh-automation/.github/workflows/dco.yml@v1
```

**Comment by FrankKr** [comment]
At OpenSTEF we experienced the same problem, disabled the DCO check (for now).

**Comment by gr2m** [comment]
can you share the exact pull request where the problem occurred?

**Comment by amzn-changml** [comment]
Hi, it appears that this issue has re-appeared as of May 6th @ 14:00 GMT. Multiple repos and orgs have stalled checks in their PRs.

**Comment by tykeal** [comment]
**A recovery mechanism for stuck checks has shipped 🎉**

This issue describes the DCO check getting stuck on **"Expected — Waiting for status to be reported."** The root cause is that if the app misses a webhook delivery (for example during an outage or restart), it never creates the check run for that PR — and if DCO is a *required* status check, the PR is then blocked indefinitely with no button in the UI to re-request it.

There is now a self-service fix. **Commenting `@dcoapp recheck` on the affected (open) pull request** will force the app to (re-)create and run the DCO check, unblocking the PR — even for PRs whose original webhook was missed. See #288 (closes #283).

- Anyone can trigger it; it only acts on open PRs and only on that exact comment.
- It requires **no new permissions** for existing installations — maintainers just need to ensure the **"Issue comment"** event is enabled on their installation (it is enabled by default going forward).

This lands alongside broader reliability work to reduce how often the app misses events in the first place (a Probot v14 upgrade and runtime fixes, plus post-deployment smoke tests). Outages can still happen, but they should now be both rarer and **recoverable** without maintainer intervention.

Closing as resolved by #288. If you hit a stuck check that `@dcoapp recheck` does **not** clear, please open a new issue with the PR link.

**Comment by tykeal** [comment]
**Correction to my earlier comment on the `@dcoapp recheck` command.**

The command shipped as a **pull request review** action, not a plain conversation comment (the comment-based version couldn't be enabled without an additional permission that would have forced every install to re-approve). On an open PR:

- Submit a **pull request review** with `@dcoapp recheck` on its own line in the review summary (Files changed → Review changes → Comment → Submit), **or**
- Add an **inline review comment** containing `@dcoapp recheck` on its own line.

Either re-runs (or first-creates) the DCO check to recover a PR stuck "waiting for status." See #289 for details. (Please open a new issue if you hit any problems.)
